src/mscorlib/shared/System/MemoryExtensions.cs

```diff
@@ -707,21 +707,82 @@ public static bool EndsWith<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> value)
         }
 
         /// <summary>
-        /// Creates a new span over the portion of the target array.
+        /// Creates a new span over the target array.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> AsSpan<T>(this T[] array)
         {
             return new Span<T>(array);
         }
 
+        /// <summary>
+        /// Creates a new Span over the portion of the target array beginning
+        /// at 'start' index and ending at 'end' index (exclusive).
```

Comment:
There is no 'end' index. We have length.

Reply:
The 'end' index phrasing device is used in the xml docs for the Span constructors which is where these comments were copy-pasted from. It makes it easier to describe the edge conditions for the exceptions.